apps/api/src/domain/models/audio-visual/shared/commands/transcripts/translate-line-item/line-item-translated.event-handler.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { EventSourcedAudioItemViewModel } from '../../../../audio-item/queries';
+import { TranscriptItem } from '../../../entities/transcript-item.entity';
+import { TranscriptParticipant } from '../../../entities/transcript-participant';
+import { Transcript } from '../../../entities/transcript.entity';
+import { LineItemTranslated } from './line-item-translated.event';
+import { LineItemTranslatedEventHandler } from './line-item-translated.event-handler';
+
+const audioItemId = buildDummyUuid(60);
+
+const translationLanguageCode = LanguageCode.English;
+
+const originalLanguageCode = LanguageCode.Chilcotin;
+
+const translationText = 'text of the translation';
+
+const participant = buildTestInstance(TranscriptParticipant);
+
+const inPoint = 1500;
+
+const outPoint = 3600;
+
+const lineItemTranslated = buildTestInstance(LineItemTranslated, {
+    payload: {
+        aggregateCompositeIdentifier: { id: audioItemId, type: AggregateType.audioItem },
+        translation: translationText,
+        languageCode: translationLanguageCode,
+        inPointMilliseconds: inPoint,
+        outPointMilliseconds: outPoint,
+    },
+});
+
+const existingAudioItem = buildTestInstance(EventSourcedAudioItemViewModel, {
+    id: audioItemId,
+    transcript: buildTestInstance(Transcript, {
+        participants: [participant],
+        items: [
+            buildTestInstance(TranscriptItem, {
+                inPointMilliseconds: inPoint,
+                outPointMilliseconds: outPoint,
+                text: buildMultilingualTextWithSingleItem(
+                    'existing transcript item text',
+                    originalLanguageCode
+                ),
+            }),
+        ],
+    }),
+});
+
+describe(`LineItemTranslatedEventHandler`, () => {
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let app: INestApplication;
+
+    let testRepositoryProvider: IQueryRepositoryProvider;
+
+    let lineItemTranslatedEventHandler: LineItemTranslatedEventHandler;
+
+    beforeAll(async () => {
+        const moduleRef = await Test.createTestingModule({
+            imports: [PersistenceModule.forRootAsync()],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        await moduleRef.init();
+
+        app = moduleRef.createNestApplication();
+
+        const connectionProvider = app.get(ArangoConnectionProvider);
+
+        databaseProvider = new ArangoDatabaseProvider(connectionProvider);
+
+        testRepositoryProvider = app.get(QUERY_REPOSITORY_PROVIDER_TOKEN);
+
+        lineItemTranslatedEventHandler = new LineItemTranslatedEventHandler(
+            // @ts-expect-error TODO find a better way to sidestep this type issue
```

Comment:
Can we explain what the issue is instead? And we should update other locations. As I recall, the problem is that the provider the handler wants is restricted to `forResource(type:  AudioVisualResourceType)` and not the full `ResourceType`. We never receive an event that has a different type due to validation at the command layer, which is responsible for emitting the corresponding events.

Reply:
done